integration-tests/deployment/memory/chain.go

```diff
@@ -58,6 +58,25 @@ func GenerateChains(t *testing.T, numChains int) map[uint64]EVMChain {
 	chains := make(map[uint64]EVMChain)
 	for i := 0; i < numChains; i++ {
 		chainID := chainsel.TEST_90000001.EvmChainID + uint64(i)
+		key, err := crypto.GenerateKey()
+		require.NoError(t, err)
+		owner, err := bind.NewKeyedTransactorWithChainID(key, big.NewInt(1337))
+		require.NoError(t, err)
+		// there have to be enough initial funds on each chain to allocate for all the nodes that share the given chain in the test
+		backend := backends.NewSimulatedBackend(core.GenesisAlloc{
+			owner.From: {Balance: big.NewInt(0).Mul(big.NewInt(7000), big.NewInt(params.Ether))}}, 50000000)
+		tweakChainTimestamp(t, backend, time.Hour*8)
+		chains[chainID] = EVMChain{
+			Backend:     backend,
+			DeployerKey: owner,
+		}
+	}
+	return chains
+}
+
+func GenerateChainsWithIds(t *testing.T, chainIDs []uint64) map[uint64]EVMChain {
```

Comment:
why specific IDs?

Reply:
it made it easier to test keystone where not all chains are equal. specifically that the capability registry and ocr 3 contracts are on the expected chain.

this may be cruft that can be ripped out, but i'm hesitant to do that now given that it's working